### PR TITLE
WTA-14: First commit for login and sign up pages

### PR DESCRIPTION
**Pull request by aneeshb7**

### CODE REVIEW

File: backend/.env.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+PORT=8080
+DATABASE_URI=mongodb+srv://admin:Password123@workoutapp.7s766.mongodb.net/myFirstDatabase?retryWrites=true&w=majority
```
Comment (aryang13) [inline backend/.env.txt]: This file should not be pushed up. Also, it shouldn't be .env.txt just env

**Review by aryang13 — COMMENTED** [review]
Remove package-lock.json from PR for backend folder since it is removing all the dependencies

**Review by aryang13 — COMMENTED** [review]
what is rnFirebaseAuth folder for?

File: backend/.env.txt (hunk 1 above)
Comment (aneeshb7) [reply backend/.env.txt]: Deleted file from PR

File: frontend/screens/SignUpPage.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    container: {
+        flex: 1,
+        backgroundColor: 'white',
+        alignItems: 'center',
+        justifyContent: 'center'
+    },
+    backButtonContainer: {
+        ...StyleSheet.absoluteFillObject,
+        top: 65,
+        right: 350
```
Comment (straw-hat-gupta) [inline frontend/screens/SignUpPage.js]: this should be right: 330 so all the letters for the button are on the same line

File: frontend/screens/SignUpPage.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
+import { NavigationContainer, PrivateValueStore } from '@react-navigation/native';
+import React from 'react';
+import { StyleSheet, Text, TextInput, View, Button, Alert } from 'react-native';
+import Login from './Login';
```
Comment (straw-hat-gupta) [inline frontend/screens/SignUpPage.js]: Not needed

File: frontend/screens/SignUpPage.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
+import { NavigationContainer, PrivateValueStore } from '@react-navigation/native';
```
Comment (straw-hat-gupta) [inline frontend/screens/SignUpPage.js]: why do you need this?

File: frontend/routes/LogInNavigation.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import { createAppContainer, createSwitchNavigator } from "react-navigation";
```
Comment (aryang13) [inline frontend/routes/LogInNavigation.js]: name the folder stacks instead of routes

File: frontend/App.js [hunk 6]
```diff
@@ -1,20 +1,11 @@
-import { StatusBar } from 'expo-status-bar';
-import { StyleSheet, Text, View } from 'react-native';
+import React from 'react';
+import Login from './screens/Login';
+import SignUpPage from './screens/SignUpPage';
+import Navigator from './routes/LogInNavigation';
```
Comment (aryang13) [inline frontend/App.js]: change name to importNavigator instead of Navigator

File: frontend/BackPressHandler.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
```
Comment (aryang13) [inline frontend/BackPressHandler.js]: Where is this file used?

**Review by aryang13 — COMMENTED** [review]
Remove the package-lock.json file from the backend as it is causing a merge conflict.

**Review by aryang13 — COMMENTED** [review]
(removed this comment)

**Review by aryang13 — COMMENTED** [review]
We can also look into separating the styling into a separate folder

File: frontend/BackPressHandler.js (hunk 7 above)
Comment (aneeshb7) [reply frontend/BackPressHandler.js]: it isnt, thought i would need it but dont for the time being so I will delete it

File: frontend/App.js (hunk 6 above)
Comment (aneeshb7) [reply frontend/App.js]: why would i change this?

**Comment by aryang13** [comment]
separate the style sheet into a separate folder and file instead of putting it in the same file as the content (https://medium.com/the-react-native-log/tips-for-styling-your-react-native-apps-3f61608655eb) and change the navigator name and so think we are good (import LoginNavigator from ....)

File: frontend/App.js [hunk 8]
```diff
@@ -1,20 +1,8 @@
-import { StatusBar } from 'expo-status-bar';
-import { StyleSheet, Text, View } from 'react-native';
+import React from 'react';
+import Navigator from './stacks/LogInNavigation';
```
Comment (aryang13) [inline frontend/App.js]: Change to 
```
import LoginNavigator from './stacks/LogInNavigator
```

File: frontend/screens/Login.js [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+                />
+            </View>
+        </View>
+    );
+}
+
+
+
+const styles = StyleSheet.create({
+    container: {
```
Comment (aryang13) [inline frontend/screens/Login.js]: Use the same styling file for both login and signup page

File: frontend/screens/Login.js [hunk 10]
```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { StyleSheet, Button, View, Text, TextInput, Alert} from 'react-native';
+
+export default function Login({ navigation }) {
```
Comment (aryang13) [inline frontend/screens/Login.js]: change file name to LoginPage.js

File: frontend/screens/SignUpPage.js [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+            </View>
+            <TextInput
+                placeholder='Re-enter Password'
+                style={styles.textBox}
+                secureTextEntry={true}
+            />
+            <View style={styles.buttonMargin}>
+                <Button // add box to button with colour and appropriate amount of padding 
+                    title="Create Account"
+                    onPress={() => Alert.alert("Make this send an API Request... this is pending")}
```
Comment (aryang13) [inline frontend/screens/SignUpPage.js:52]: create a separate function like we did for backPressHandler

File: frontend/screens/Login.js [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+
+            <TextInput 
+            placeholder= 'Password'
+            style={styles.input}
+            secureTextEntry={true}
+            />  
+            <View styles={styles.buttonContainer}>
+                <Button 
+                    title= 'Login'
+                    onPress={() => Alert.alert("Not yet hoe")}
```
Comment (aryang13) [inline frontend/screens/Login.js]: create separate handler function

File: frontend/App.js (hunk 8 above)
Comment (aneeshb7) [reply frontend/App.js]: This is actually creating a lot of issues but I will figure out how to fix it

File: frontend/screens/SignUpPage.js (hunk 4 above)
Comment (aneeshb7) [reply frontend/screens/SignUpPage.js]: Removed that, you were right didn't need it

File: frontend/screens/SignUpPage.js (hunk 3 above)
Comment (aneeshb7) [reply frontend/screens/SignUpPage.js]: Removed

File: frontend/screens/SignUpPage.js (hunk 2 above)
Comment (aneeshb7) [reply frontend/screens/SignUpPage.js]: I'll test it out on my machine and push that if it looks better

File: frontend/screens/Login.js (hunk 9 above)
Comment (aneeshb7) [reply frontend/screens/Login.js]: Sure I'll create that

File: frontend/screens/Login.js (hunk 10 above)
Comment (aneeshb7) [reply frontend/screens/Login.js]: Done

File: frontend/screens/SignUpPage.js (hunk 11 above)
Comment (aneeshb7) [reply frontend/screens/SignUpPage.js:52]: Is the API ready to be implemented?

File: frontend/screens/Login.js (hunk 12 above)
Comment (aneeshb7) [reply frontend/screens/Login.js]: As above, is the API ready to be implemented into the frontend?

File: frontend/App.js (hunk 8 above)
Comment (aryang13) [reply frontend/App.js]: https://smartlogic.io/blog/basic-react-navigation-for-react-native-apps/

File: frontend/screens/Login.js (hunk 12 above)
Comment (aryang13) [reply frontend/screens/Login.js]: we can implement later but separate it for now

File: frontend/screens/SignUpPage.js (hunk 11 above)
Comment (aryang13) [reply frontend/screens/SignUpPage.js:52]: we can implement later but separate it for now

**Review by aryang13 — APPROVED** [review]

**Pull request merged**